buildspecs/pr-build-amzn.yml

```diff
@@ -0,0 +1,120 @@
+version: 0.2
+
+env:
+  variables:
+    # Github username of the forked repo on which to make builds
+    GITHUBUSERNAME: aws
+
+phases:
+  install:
+    commands:
+      # Same buildspec used for different Amazon Linuc versions - detect verison to use in Go installation
+      - |
+        if [ -f /etc/os-release ]; then
+          . /etc/os-release
+          case $ID:$VERSION_ID in
+            amzn:2)
+              amzn_version="amzn2"
+              ;;
+            amzn:2023)
+              amzn_version="amzn2023"
+              ;;
+            *)
+              echo "Unsupported Linux distribution: $ID:$VERSION_ID"
+              exit 1
+              ;;
+          esac
+        else
+          echo "Unable to detect the Linux distribution"
+          exit 1
+        fi
+
+      # Same buildspec for different architectures - detect the architecture here and rename the artifacts accordingly
+      - architecture=""
+      - case $(uname -m) in
+           x86_64)
+             architecture="amd64"
+           ;;
+           aarch64)
+             architecture="arm64"
+           ;;
+        esac
+
+      # Identify the correct AL2023 version to use in the Go installation
+      - AL23_VERSION="$(curl -s https://al2023-repos-us-west-2-de612dc2.s3.dualstack.us-west-2.amazonaws.com/core/releasemd.xml | xmllint --xpath "string(//root/releases/release[last()]/@version)" -)"
```

Comment:
Can you add a comment like: https://github.com/aws/amazon-ecs-ami/blob/f7a406dcea7467061452a6da77b5543ff25d4c46/generate-release-vars.sh#L95-L97

incase this stops working, we would know where to look for the right source URL.

Reply:
Yes, great idea.